User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs

```diff
@@ -0,0 +1,388 @@
+                };
+                var reader = new ShimDataReader();
+                var writer = new StubDataWriter<double>();
+                CollectionSortHandler<double> collectionSortHandler;
+                bool result;
+
+                //
+                // Act.
+                //
+                collectionSortHandler = new CollectionSortHandler<double>(reader, writer, parser);
+                collectionSortHandler.GenerateSorter(sorterType);
+                collectionSortHandler.Read(path);
+                collectionSortHandler.Execute();
+                result = collectionSortHandler.SortedCollection.SequenceEqual(expectedCollection);
+
+                //
+                // Assert.
+                //
+                Assert.IsTrue(result);
+            }
+        }
+
+        /// <summary>
+        /// Tests Read if pass correct path to the file it will get correct content of the file.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_Read_PassCorrectPathToTheFile_GetCorrectContent()
+        {
+            using (ShimsContext.Create())
+            {
+                //
+                // Arrange.
+                //
+                int[] actualCollection;
+                var expectedCollection = new int[] { 125, 12, 33 };
+                var parser = new StubIDataParser<int>()
+                {
+                    ConvertDataString = (str) => expectedCollection
+                };
+                var path = @"D:\collectionToRead.txt";
+                var reader = new ShimDataReader();
+                var writer = new StubDataWriter<int>();
+                CollectionSortHandler<int> collectionSortHandler;
+
+                //
+                // Act.
+                //
+                collectionSortHandler = new CollectionSortHandler<int>(reader, writer, parser);
+                collectionSortHandler.Read(path);
+                actualCollection = collectionSortHandler.UnSortedCollection;
+
+                //
+                // Assert.
+                //
+                Assert.AreEqual(expectedCollection, actualCollection);
+            }
+        }
+
+        /// <summary>
+        /// Tests Read if pass empty content to parser it will throw ArgumentNullException.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_Read_PassEmptyContentToParser_ThrowArgumentNullException()
+        {
+            using (ShimsContext.Create())
+            {
+                //
+                // Arrange.
+                //
+                CollectionSortHandler<int> collectionSortHandler;
+                var path = @"D:\collectionToRead.txt";
+                var parser = new StubIDataParser<int>()
+                {
+                    ConvertDataString = (str) => throw new ArgumentNullException()
```

Comment:
Here are two asserts for the single test method. 
Should be also separated.

Reply:
I've slightly changed test case in Arrange part and changed name please check again.